### ISSUE TITLE
Bug: Product manifold is not able to project if one of the components is `Euclidean`

### ISSUE THREAD

**Issue by miguelgondu** [geomstats/geomstats#1655]
### Describe the bug

If one of the components of a product of manifolds is $\mathbb{R}^n$, the implementation of `projection` throws an error at `is_intrinsic`.

More specifically, `is_intrinsic` in `ProductRiemannianMetric` is implemented like this:

```python
    def is_intrinsic(self, point):
        """..."""
        if self.default_point_type != "vector":
            raise ValueError("Invalid default_point_type: 'vector' expected.")

        if point.shape[-1] == self.dim:
            return True
        if point.shape[-1] == sum(dim + 1 for dim in self.dims):
            return False
        raise ValueError("Input shape does not match the dimension of the manifold")
```

That is, it is expecting *all* of the manifolds to be either intrinsic or extrinsic. This means that I cannot combine $\mathbb{R}^n$ with other manifolds that I want to define as extrinsic.

Also, if I read it correctly, we are assuming that if a point is not intrinsic, then it must be embedded in $\mathbb{R}^{\text{dim}(M) + 1}$. Isn't this problematic if the ambient space is actually higher-dimensional?

I include a small example using `torch` as a backend, and the manifolds $\mathbb{S}^2\times \mathbb{S}^2$ and $\mathbb{R}^2\times \mathbb{S}^2$.

### Steps/Code to Reproduce

```python
import torch

from geomstats.geometry.hypersphere import Hypersphere
from geomstats.geometry.euclidean import Euclidean
from geomstats.geometry.product_manifold import ProductManifold

r2_s2 = ProductManifold([Euclidean(2), Hypersphere(2)])
s2_s2 = ProductManifold([Hypersphere(2), Hypersphere(2)])

# Some point almost on the product R2 x S2
point_to_be_projected_to_r2_s2 = torch.Tensor([0.0, 0.0] + [1.01, 0.0, 0.0])

# Some point almost on the product S2 x S2
point_to_be_projected_to_s2_s2 = torch.Tensor([1.01, 0.0, 0.0] + [1.01, 0.0, 0.0])

# Trying to project to S2 x S2 works wonderfully...
s2_s2_projection = s2_s2.projection(point_to_be_projected_to_s2_s2)
print(f"Projection to S2xS2: {s2_s2_projection}")

# ...but projecting to R2 x S2 fails.
r2_s2_projection = r2_s2.projection(point_to_be_projected_to_r2_s2)
print(f"Projection to R2xS2: {r2_s2_projection}")
```

### Expected Behaviour

No error is thrown, and the console prints

```
Projection to S2xS2: tensor([1., 0., 0., 1., 0., 0.])
Projection to R2xS2: tensor([0., 0., 1., 0., 0.])
```

### Actual Behaviour

An error is thrown.

```
Exception has occurred: ValueError
Input shape does not match the dimension of the manifold

File "/home/thisisprivate/anaconda3/envs/env/lib/python3.9/site-packages/geomstats/geometry/product_riemannian_metric.py", line 118, in is_intrinsic
    raise ValueError("Input shape does not match the dimension of the manifold")
File "/home/thisisprivate/anaconda3/envs/env/lib/python3.9/site-packages/geomstats/geometry/product_manifold.py", line 229, in projection
    intrinsic = self.metric.is_intrinsic(point)
File "/home/thisisprivate/some_file.py", line 21, in <module>
    r2_s2_projection = r2_s2.projection(point_to_be_projected_to_r2_s2)
```

in line 118 of `product_riemannian_metric.py`.

### Your environment

```shell
- `geomstats==2.5.0`
- `torch==1.12.1`
- `GEOMSTATS_BACKEND='pytorch'`
- Ubuntu 20.04.
```
